src/main/java/de/tum/in/www1/artemis/service/ZipFileService.java

```diff
@@ -79,4 +85,57 @@ private void copyToZipFile(ZipOutputStream zipOutputStream, Path path, ZipEntry
             log.error("Create zip file error", e);
         }
     }
+
+    /**
+     * Extracts a zip file recursively.
+     *
+     * @param zipFile The path to the zip file
+     * @throws IOException if something goes wrong
+     */
+    public void extractZipFileRecursively(String zipFile) throws IOException {
```

Comment:
this code is only used in tests, so please move it into a service in the test folder